scripts/release.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+echo "enter release string according to semantic versioning (e.g. v1.2.3)."
+read -r INPUT
+if [[ ! "${INPUT}" =~ ^v[0-9]+.[0-9]+.[0-9]+ ]]; then
+  echo "Expected 'version' param of the form 'v<major-version>.<minor-version>.<patch-version>' but got '${INPUT}'"
+  exit 1
+fi
+
+VERSION=${INPUT#v}
+RELEASE_VERSION="${VERSION}"
+MINOR_VERSION=$(echo "${VERSION}" | cut -d. -f 1-2)
+
+REPOSITORY=${REPOSITORY:-"git@github.com:etcd-io/bbolt.git"}
+REMOTE="${REMOTE:-"origin"}"
+
+remote_tag_exists=$(git ls-remote --tags "${REPOSITORY}" | grep -c "${INPUT}" || true)
+if [ "${remote_tag_exists}" -gt 0 ]; then
+   echo "Release version tag exists on remote. Checking out refs/tags/${INPUT}"
+   git checkout -q "tags/${INPUT}"
```

Comment:
Can we do an exit here rather than continuing? If the tag exists, it will try to update the `version.go` file and will try to create the tag again.